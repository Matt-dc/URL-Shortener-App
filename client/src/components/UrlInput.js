import React from "react";
import { InputGroup, FormControl, Button } from "react-bootstrap";
import { CopyToClipboard } from "react-copy-to-clipboard";

import Spinner from "./Spinner";

export default function ShortenerInput(props) {
  const changeHandler = e => {
    props.handleChange(e);
  };

  const spinner = <Spinner className="spinner" size="2x" spinning="spinning" />;

  return (
    <InputGroup size="lg" className="mb-3">
      <FormControl
        style={{ color: props.shortened ? "#5f82d1" : "" }}
        onChange={changeHandler}
        value={props.URL}
        placeholder="Paste in your link"
      />

      <InputGroup.Append>
        {props.shortened ? (
          <CopyToClipboard
            text={props.URL}
            onCopy={() => props.setCopiedTrue("main")}
          >
            <Button variant={props.copied === "main" ? "success" : "primary"}>
              {props.copied === "main" ? "Copied" : "Copy"}
            </Button>
          </CopyToClipboard>
        ) : (
          <Button
            className="shorten-btn-style"
            disabled={props.disabled}
            onClick={props.handleShorten}
          >
            {props.loading ? spinner : "Shorten"}
          </Button>
        )}
      </InputGroup.Append>
    </InputGroup>
  );
}
