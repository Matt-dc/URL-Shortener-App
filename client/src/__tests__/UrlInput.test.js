import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, mount, configure } from "enzyme";
import sinon from "sinon";
import UrlInput from "../components/UrlInput";

configure({ adapter: new Adapter() });

describe("<UrlInput />", () => {
  it("Should check component mounts", () => {
    const wrapper = shallow(<UrlInput />);
    expect(wrapper.find(".url-input").exists());
    expect(wrapper).toMatchSnapshot();
  });

  it("Checks if the shorten button is rendered", () => {
    const wrapper = shallow(<UrlInput />);
    expect(wrapper.find("Button").length).toEqual(1);
  });

  it("calls handleShorten on click", () => {
    const spy = sinon.spy();
    const wrapper = mount(<UrlInput handleShorten={spy} />);

    wrapper
      .find("Button")
      .first()
      .simulate("click");

    expect(spy.calledOnce).toBe(true);
  });
});
