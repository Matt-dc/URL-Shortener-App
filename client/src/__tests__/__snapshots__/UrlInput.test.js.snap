// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UrlInput /> Should check component mounts 1`] = `ShallowWrapper {}`;
